dsc/utilities/aws/s3.py

```diff
@@ -58,23 +58,51 @@ def put_file(
         return response
 
     def get_files_iter(
-        self, bucket: str, file_type: str, excluded_key_prefix: str
+        self,
+        bucket: str,
+        prefix: str = "",
+        file_identifier: str = "",
+        file_type: str = "",
+        exclude_prefixes: list[str] | None = None,
```

Comment:
Could this just be `exclude_prefixes: list[str] = []`? You could remove lines 90-91:

```
if exclude_prefixes is None:
   exclude_prefixes = []
```

Reply:
Ah, did not know that, thank you for the heads up!